### PR TITLE
Add support for multiple signing keys and domains

### PR DESCRIPTION
**Pull request by Supermathie**
* complete rework of the mechanism for passing data into the container
  * keys themselves are now passed by ENV
  * multiple domains can be provided for signing
* the extraction of this information is now handled by a ruby script

### CODE REVIEW

**Comment by saj** [comment]
Reviewer acknowledges intentional break in backwards compat.

It seems awkward to define `DKIM_DEFAULT_DOMAIN` without a corresponding configuration knob for _default key_.  In practice, the set of sender domains is always (AFAIK?) known in advance.  (Operators must garnish these zones with the requisite `_domainkey` TXT RRs, after all.)  If permitted by `opendkim`, I think it would be better to drop the default entirely.

The `KeyTable` would look the same:

```
foo.net foo.net:foo:/whatever/keys/foo.net.pem
bar.net bar.net:bar:/whatever/keys/bar.net.pem
```

But the `SigningTable` would have no fallback:

```
*@foo.net foo.net
*@bar.net bar.net
# stop here - no unguarded wildcard (*)
```

If presented with a message `From:` some other domain, like `baz.net`, correct behaviour would be for `opendkim` to echo the message through the milter protocol without any modifications -- it **must not** insert the `DKIM-Signature:` header.  Is this possible?

**Comment by Supermathie** [comment]
> Reviewer acknowledges intentional break in backwards compat.

Yes; I think I'll add some guards around that… *or* I could keep backwards compatibility. Eh, that's probably not worth it…

> It seems awkward to define DKIM_DEFAULT_DOMAIN without a corresponding configuration knob for default key.

A domain has two configuration pieces key and selector. In theory, you could use *multiple* signing keys for a single domain in the config, but we don't do that here.

The SigningTable is used to look up the entry in the KeyTable to use; there's no default for KeyTable the same way there's no default for SigningTable.

> I think it would be better to drop the default entirely.

I was thinking along similar lines; I'm going to remove the *requirement* for the default which will allow us to still mimic previous behaviour.

**Review by saj — APPROVED** [review]
LGTM.  @david, we'd appreciate a Ruby correctness and readability review if you can spare a moment.

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # libmilter in alpine is IPv4 only...
-FROM debian:buster-slim
+FROM ruby:2.7.2-slim-buster
```
Comment (saj) [inline Dockerfile:2]: Oh, one thing.  Could we stick to `debian`?

https://github.com/discourse/mail-receiver/pull/6#issuecomment-779780391

**Review by davidtaylorhq — APPROVED** [review]
Ruby all looks good to me 👌

File: Dockerfile (hunk 1 above)
Comment (Supermathie) [reply Dockerfile:2]: Done!

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**